gtrak/gtreveroot.F

```diff
@@ -78,20 +78,35 @@ SUBROUTINE GTREVEROOT
       MTRACK=-999
       FINISHPRIMA=0
 10    ISPRIMA=0
-      IF(FINISHPRIMA.GE.1) THEN
+* Finish previous primary immediately
+      IF(FINISHPRIMA.GE.2) THEN
          CALL RXOUTH
          FINISHPRIMA=0
       ENDIF
       CALL RXGTRAK(MTRACK,IPART,PMOM,E,VPOS,VPOLA,TTOF,ISPRIMA)
-* Finish primary track if flagged
+      IF(FINISHPRIMA.GE.1) THEN
+* Finish primary only when the next one has been popped...
+         IF(ISPRIMA.GE.1) THEN
+            CALL RXOUTH
+         ENDIF
+* ...or when the stack is empty
+         IF(MTRACK.LE.0) THEN
+            CALL RXOUTH
```

Comment:
This causes calling FinishPrimary at the end of event twice. However fixing this double call does not solve the problem in digitization on my side.

Reply:
But it happens. If you add a printing in FinishPrimary, you will see it printed 2x at the end of event. It can be a side effect of  returning a particle even when  mtrack=-1 by AliStack.

Reply:
Interesting... I will let you know. But in any case, I will revise the track transfer now and the way in which the `TMCManagerStack`s are worked with to have the very same order of popping as before